MSAL/test/integration/native_auth/end_to_end/sign_up/MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests.swift

```diff
@@ -52,10 +51,15 @@ final class MSALNativeAuthSignUpUsernameAndPasswordEndToEndTests: MSALNativeAuth
 
         // Now submit the code...
```

Comment:
can we check if the code arrived only if `delegate.onSignUpCodeRequiredCalled` is true, as we did in the other test classes? So, we could check if a code arrived only when we're in the right state. 
This comment applies for all tests in this class.

Reply:
Done, thanks. Do you think we should stop the test if any of the other delegates doesn't get called? (such as `signUpVerifyCodeDelegate.onSignUpPasswordRequiredCalled` or `signUpAttributesRequiredDelegate.onSignUpAttributesRequiredErrorCalled`)

IMO yes, but this would affect the other sign up with code PR too FYI @borgesmb

Reply:
Yes, I think we should. At least this is what we do in other tests like signIn with code. I thought it was the same for signUp too

Reply:
Done